File: MP.Web/ClientApp/src/app/services/playlist/playlist.service.ts
import { Injectable, Inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { BASE_URL } from '@mintplayer/ng-base-url';
import { PaginationRequest, PaginationResponse } from '@mintplayer/ng-pagination';
import { Playlist } from '../../entities/playlist';
import { ePlaylistScope } from '../../enums/ePlaylistScope';

@Injectable({
  providedIn: 'root'
})
export class PlaylistService {
  constructor(private httpClient: HttpClient, @Inject(BASE_URL) private baseUrl: string, @Inject('API_VERSION') private apiVersion: string) {
  }

  public pagePlaylists(request: PaginationRequest, scope: ePlaylistScope) {
    switch (scope) {
      case ePlaylistScope.My:
        return this.httpClient.post<PaginationResponse<Playlist>>(`${this.baseUrl}/web/${this.apiVersion}/playlist/my/page`, request).toPromise();
      case ePlaylistScope.Public:
        return this.httpClient.post<PaginationResponse<Playlist>>(`${this.baseUrl}/web/${this.apiVersion}/playlist/public/page`, request).toPromise();
    }
  }

  public getPlaylists(scope: ePlaylistScope, include_relations: boolean = false) {
    let headers = {
      'include_relations': String(include_relations)
    };

    switch (scope) {
      case ePlaylistScope.My:
        return this.httpClient.get<Playlist[]>(`${this.baseUrl}/web/${this.apiVersion}/playlist/my`, { headers }).toPromise();
      case ePlaylistScope.Public:
        return this.httpClient.get<Playlist[]>(`${this.baseUrl}/web/${this.apiVersion}/playlist/public`, { headers }).toPromise();
    }
  }

  public getPlaylist(id: number, include_relations: boolean = false) {
    return this.httpClient.get<Playlist>(`${this.baseUrl}/web/${this.apiVersion}/playlist/${id}`, {
      headers: {
        'include_relations': String(include_relations)
      }
    }).toPromise();
  }

  public createPlaylist(playlist: Playlist) {
    return this.httpClient.post<Playlist>(`${this.baseUrl}/web/${this.apiVersion}/playlist`, playlist).toPromise();
  }

  public updatePlaylist(playlist: Playlist) {
    return this.httpClient.put<Playlist>(`${this.baseUrl}/web/${this.apiVersion}/playlist/${playlist.id}`, playlist).toPromise();
  }

  public deletePlaylist(playlist: Playlist) {
    return this.httpClient.delete<Playlist>(`${this.baseUrl}/web/${this.apiVersion}/playlist/${playlist.id}`).toPromise();
  }
}

File: MintPlayer.Data/Services/PlaylistService.cs
using MintPlayer.Dtos.Dtos;
using MintPlayer.Data.Repositories;
using MintPlayer.Pagination;
using MintPlayer.Data.Enums;
using MintPlayer.Data.Abstractions.Services;
using MintPlayer.Data.Abstractions.Enums;

namespace MintPlayer.Data.Services;

internal class PlaylistService : IPlaylistService
{
	private readonly IPlaylistRepository playlistRepository;
	public PlaylistService(IPlaylistRepository playlistRepository)
	{
		this.playlistRepository = playlistRepository;
	}

	public async Task<PaginationResponse<Playlist>> PagePlaylists(PaginationRequest<Playlist> request, EPlaylistScope playlistScope)
	{
		var playlists = await playlistRepository.PagePlaylists(request, playlistScope);
		return playlists;
	}

	public async Task<IEnumerable<Playlist>> GetPlaylists(EPlaylistScope playlistScope, bool include_relations = false)
	{
		var playlists = await playlistRepository.GetPlaylists(playlistScope, include_relations);
		return playlists;
	}

	public async Task<Playlist> GetPlaylist(int id, bool include_relations = false)
	{
		var playlist = await playlistRepository.GetPlaylist(id, include_relations);
		return playlist;
	}

	public async Task<Playlist> InsertPlaylist(Playlist playlist)
	{
		var new_playlist = await playlistRepository.InsertPlaylist(playlist);
		return new_playlist;
	}

	public async Task<Playlist> UpdatePlaylist(Playlist playlist)
	{
		var updated_playlist = await playlistRepository.UpdatePlaylist(playlist);
		return updated_playlist;
	}

	public async Task DeletePlaylist(int playlistId)
	{
		await playlistRepository.DeletePlaylist(playlistId);
		await playlistRepository.SaveChangesAsync();
	}

	public async Task SaveChangesAsync()
	{
		await playlistRepository.SaveChangesAsync();
	}
}
